terraform/gen/plural_resource.go.tpl

```diff
@@ -116,9 +119,22 @@ func (r resourceTeleport{{.Name}}) Create(ctx context.Context, req tfsdk.CreateR
 {{- else }}
 	{{.VarName}}Resource := {{.VarName}}
 {{end}}
+
+{{- if .ForceSetKind }}
+	{{.VarName}}Resource.Kind = {{.ForceSetKind}}
+{{- end}}
+
+{{if .HasCheckAndSetDefaults -}}
+	err = {{.VarName}}Resource.CheckAndSetDefaults()
+	if err != nil {
+	resp.Diagnostics.Append(diagFromWrappedErr("Error setting {{.Name}} defaults", trace.Wrap(err), "{{.Kind}}"))
+	return
+	}
+{{- end}}
+
```

Comment:
`CheckAndSetDefaults` and kind setting were moved before

```
id := {{.VarName}}Resource.Metadata.Name
```

because the name itself can be changed during this step (e.g. nameless openssh servers).